### PR TITLE
Add Node 18 support

### LINKED ISSUE #3214: Node 18 compatibility
### Is your enhancement related to a problem? Please describe.

Currently, running `npm run build` in a node 18 env throws an error. It would be great if we could update 10up-toolkit and apply [this workaround](https://github.com/10up/10up-toolkit/issues/246#issuecomment-1306068726) so the plugin becomes compatible with node 18.

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by burhandodhy**
### Description of the Change

This PR adds the support for Node 18 by updating the 10up-toolkit to latest version and added the `eslint-plugin-jsdoc` in `overrides`

Closes #3214 

### How to test the Change

1. Switch the node version to 18
2. Run `npm run build` command

### Changelog Entry

> Added - Node 18 support

### Credits

Props @burhandodhy @felipeelia 

### Checklist:

- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: .stylelintrc [hunk 1]
```diff
@@ -11,7 +11,13 @@
...
     ],
     "font-family-no-missing-generic-family-keyword": null,
-    "declaration-property-unit-allowed-list": null
+    "declaration-property-unit-allowed-list": null,
+    "value-keyword-case": [
+      "lower",
+      {
+        "camelCaseSvgKeywords": true
+      }
+    ]
```
Comment (felipeelia) [inline .stylelintrc]: @burhandodhy why is this needed?

**Review by felipeelia — CHANGES_REQUESTED** [review]
This looks good @burhandodhy but before we merge it, can we please also bump the node version in our GitHub actions? Thanks!

File: .stylelintrc (hunk 1 above)
Comment (burhandodhy) [reply .stylelintrc]: @felipeelia style lint gives the error for the svg "currentColor" property `Expected "currentColor" to be "currentcolor"  value-keyword-case`. This rule allows the property to be the camelCase
